src/views/app.view.tsx

```diff
@@ -1,29 +1,98 @@
 import React from "react";
 import { Switch, Redirect } from "react-router-dom";
 import { connect } from "react-redux";
-import PropTypes from "prop-types";
 import { useTheme } from "react-jss";
 
-import useAppStyles from "./app.styles";
-import routes from "../routing/routes";
-import * as globalActions from "../store/global/global.actions";
-import * as globalThunks from "../store/global/global.thunks";
-import Spinner from "./shared/spinner/spinner.view";
-import { COORDINATOR_STATE_REFRESH_RATE, RETRY_POOL_TXS_RATE } from "../constants";
-import Route from "./shared/route/route.view";
-import BaseLayout from "./shared/base-layout/base-layout.view";
-import { closeSnackbar } from "../store/global/global.actions";
-import UnderMaintenanceError from "./shared/under-maintenance-error/under-maintenance-error.view";
+import { AppDispatch, AppState, AppAction } from "src/store";
+import * as globalThunks from "src/store/global/global.thunks";
+import { closeSnackbar } from "src/store/global/global.actions";
+import routes from "src/routing/routes";
+import Spinner from "src/views/shared/spinner/spinner.view";
+import Route from "src/views/shared/route/route.view";
+import BaseLayout from "src/views/shared/base-layout/base-layout.view";
+import UnderMaintenanceError from "src/views/shared/under-maintenance-error/under-maintenance-error.view";
+import useAppStyles from "src/views/app.styles";
+import { COORDINATOR_STATE_REFRESH_RATE, RETRY_POOL_TXS_RATE } from "src/constants";
+import { AsyncTask } from "src/utils/types";
+import { Theme } from "src/styles/theme";
+//domain
+import {
+  FiatExchangeRates,
+  HermezWallet,
+  HermezNetworkStatus,
+  HermezStatus,
+} from "src/domain/hermez";
+import { EthereumNetwork } from "src/domain/ethereum";
+
+declare const window: Window &
+  typeof globalThis & {
+    ethereum?: {
+      on: (event: "accountsChanged" | "chainChanged", callback: () => void) => void;
```

Comment:
I would remove this as we are already declaring the `Window` type in the `react-app-env.d.ts`.

Reply:
To add support for the `ethereum` property we can just do this in the global scope:

```ts
import {BaseProvider} from "@ethersproject";

declare const window: Window & { ethereum?: BaseProvider };
```

Reply:
Done